src/snowflake/snowpark/modin/pandas/groupby.py

```diff
@@ -188,13 +189,28 @@ def sem(self, ddof=1):
 
     def value_counts(
         self,
-        subset=None,
+        subset: Optional[list[str]] = None,
         normalize: bool = False,
         sort: bool = True,
         ascending: bool = False,
         dropna: bool = True,
     ):
-        ErrorMessage.method_not_implemented_error(name="value_counts", class_="GroupBy")
+        query_compiler = self._query_compiler.groupby_value_counts(
+            by=self._by,
+            axis=self._axis,
+            groupby_kwargs=self._kwargs,
+            subset=subset,
+            normalize=normalize,
+            sort=sort,
+            ascending=ascending,
+            dropna=dropna,
+        )
+        if self._as_index:
```

Comment:
what sets `_as_index`?

Reply:
It's a property on the GroupBy frontend object ([here](https://github.com/snowflakedb/snowpark-python/blob/4bcd987af779e9281ff4b5bbdd37de32911ae299/src/snowflake/snowpark/modin/pandas/groupby.py#L1085)), which reads a kwarg set in GroupBy's constructor ([here](https://github.com/snowflakedb/snowpark-python/blob/4bcd987af779e9281ff4b5bbdd37de32911ae299/src/snowflake/snowpark/modin/pandas/groupby.py#L119-L124)).